pkg/cloudprovider/plugins/azure/azure_nsg_rules.go

```diff
@@ -427,7 +427,10 @@ func findSecurityRule(ruleList []*armnetwork.SecurityRule, rule *armnetwork.Secu
 func normalizeAzureSecurityRule(rule *armnetwork.SecurityRule) *armnetwork.SecurityRule {
 	property := *rule.Properties
 	normalizedProtocolNum := azureProtoNameToNumMap[strings.ToLower(string(*rule.Properties.Protocol))]
-	normalizedProtocol := protoNumAzureNameMap[normalizedProtocolNum]
+	normalizedProtocol, ok := protoNumAzureNameMap[normalizedProtocolNum]
+	if !ok {
+		normalizedProtocol = "*"
```

Comment:
we should have specific mapping of "*" to "*" in conversion. For not found cases we should give an info log as it means a user rule is in nephe priority range with nephe description. (This func's caller already did priority and description check.) We can also try to remove this user rule by returning nil here and let the caller check nil and remove the rule